sympy/core/expr.py

```diff
@@ -781,7 +781,11 @@ def _eval_interval(self, x, a, b):
         else:
             A = self.subs(x, a)
             if A.has(S.NaN, S.Infinity, S.NegativeInfinity, S.ComplexInfinity):
-                A = limit(self, x, a)
+                if a.__lt__(b):
```

Comment:
@asmeurer, @jksuom  silly question, i am still learning. what you mean guard against symbolic a < b ? if that has to do with symbols used for a and b (pi for example) shouldn't we modify `__lt__ `and `__gt__` to check symbols as well? How does `if (a < b) is not False` do that?

Thanks